src/Kestrel.Core/Internal/Http2/Http2Connection.cs

```diff
@@ -264,6 +262,11 @@ private Task ProcessHeadersFrameAsync<TContext>(IHttpApplication<TContext> appli
                 throw new Http2ConnectionErrorException(Http2ErrorCode.PROTOCOL_ERROR);
             }
 
+            if (_incomingFrame.HeadersHasPriority && _incomingFrame.HeadersStreamDependency == _incomingFrame.StreamId)
```

Comment:
What's the relation between priority and dependency?

Reply:
Priority tells the server how much of its resources should be dedicated to a given stream. Dependencies tell the server what streams to work on first.

Reply:
And here `HeadersHasPriority` checks for the priority flag on the frame. When its set on a HEADERS frame, that means the frame contains both priority and stream dependency on it (you can't send just one of those values).